ringpop.go

```diff
@@ -213,6 +233,20 @@ func (rp *Ringpop) RegisterListener(l EventListener) {
 	rp.listeners = append(rp.listeners, l)
 }
 
+// getState gets the state of the current Ringpop instance.
+func (rp *Ringpop) getState() state {
+	rp.stateMutex.Lock()
+	defer rp.stateMutex.Unlock()
```

Comment:
and not use defer. calls to `getState` are littered throughout the code and defer has serious performance impact.

```go
rp.stateMutex.RLock()
r := rp.state
rp.stateMutex.RUnlock()
return r
```